packages/react/src/Overlay/Overlay.stories.tsx

```diff
@@ -0,0 +1,403 @@
+import React, {useState, useRef, type ComponentProps} from 'react'
+import type {Args, Meta} from '@storybook/react'
+import {XIcon} from '@primer/octicons-react'
+import {Button, Text, useFocusTrap, Box, IconButton} from '..'
+import Overlay from '../Overlay'
+
+export default {
+  title: 'Private/Components/Overlay',
+  component: Overlay,
+  argTypes: {
```

Comment:
Documented the props here because something about [this merge type ](https://github.com/primer/react/blob/main/packages/react/src/Overlay/Overlay.tsx#L121) messes with how storybook reads docs 🤷🏽‍♀️